tests/python/pants_test/backend/jvm/tasks/test_scalafmt.py

```diff
@@ -165,3 +167,76 @@ def test_output_dir(self):
       relative_test_file = fast_relpath(self.test_file, self.build_root)
       with open(os.path.join(output_dir, relative_test_file), 'r') as fp:
         self.assertNotEqual(self.test_file_contents, fp.read())
+
+  def _execute_native_image(self, num_sources=1, **kwargs):
+    self.set_options(skip=False, **kwargs)
+    self.set_options_for_scope('scalafmt', use_native_image=True)
+
+    uuid = '_' + str(uuid4()).replace('-', '_')
```

Comment:
Why do we introduce randomness into these tests? I'd rather either use a constant, or `num_sources`, so that when we re-run the tests we expect identical code to run.